### PR TITLE
Add Template Network

### PR DESCRIPTION
**Pull request by AurevoirXavier**
1. For future testing.
2. Speed up integration testing CI.
3. Easier to maintain the integration testing.

By adding this new network, I review the old code and make some improvements. So this PR also contains some code cleaning.

```
     Running `target/debug/drml --chain template --alice --tmp`
2021-12-01 23:45:03 Darwinia Runtime Module Library    
2021-12-01 23:45:03 ✌️  version 2.7.0-24a0d656-x86_64-linux-gnu    
2021-12-01 23:45:03 ❤️  by Darwinia Network <hello@darwinia.network>, 2018-2021    
2021-12-01 23:45:03 📋 Chain specification: Template    
2021-12-01 23:45:03 🏷 Node name: Alice    
2021-12-01 23:45:03 👤 Role: AUTHORITY    
2021-12-01 23:45:03 💾 Database: RocksDb at /tmp/substrate9P3fgF/chains/template/db    
2021-12-01 23:45:03 ⛓  Native runtime: Template-27000 (Template-0.tx0.au0)    
2021-12-01 23:45:04 🔨 Initializing Genesis block/state (state: 0x1385…4a17, header-hash: 0x7302…45bf)    
2021-12-01 23:45:04 Using default protocol ID "sup" because none is configured in the chain specs    
2021-12-01 23:45:04 🏷 Local node identity is: 12D3KooWAv4iMJm3NsBdTdZG4w4DzdddWdL4fqSRa8Dgj8M29YMY    
2021-12-01 23:45:07 📦 Highest known block at #0    
2021-12-01 23:45:07 〽️ Prometheus server started at 127.0.0.1:9615    
2021-12-01 23:45:07 Listening for new connections on 127.0.0.1:9944.    
2021-12-01 23:45:07 Manual Seal Ready    
2021-12-01 23:45:12 💤 Idle (0 peers), best: #0 (0x7302…45bf), finalized #0 (0x7302…45bf), ⬇ 0 ⬆ 0     
```

### CODE REVIEW

File: node/cli/src/command.rs [hunk 1]
```diff
@@ -62,21 +58,26 @@ impl SubstrateCli for Cli {
...
 	fn load_spec(&self, id: &str) -> Result<Box<dyn ChainSpec>, String> {
 		let id = if id == "" {
 			let n = get_exec_name().unwrap_or_default();
-			["pangolin", "pangoro"]
+			["template", "pangolin", "pangoro"]
 				.iter()
 				.cloned()
 				.find(|&chain| n.starts_with(chain))
-				.unwrap_or("pangolin")
+				.unwrap_or("pangoro")
```
Comment (AsceticBear) [inline node/cli/src/command.rs:80]: Why change `pangolin` to `pangoro` here?

File: node/cli/src/command.rs [hunk 2]
```diff
@@ -89,15 +90,17 @@ impl SubstrateCli for Cli {
...
+			"template" | "template-dev" => Box::new(template_chain_spec::development_config()),
 			path => {
 				let path = PathBuf::from(path);
 				let chain_spec = Box::new(PangolinChainSpec::from_json_file(path.clone())?)
 					as Box<dyn ChainSpec>;
 
-				if self.run.force_pangoro || chain_spec.is_pangoro() {
-					Box::new(PangoroChainSpec::from_json_file(path)?)
-				} else {
+				if self.run.force_pangoro || chain_spec.is_pangolin() {
```
Comment (AsceticBear) [inline node/cli/src/command.rs:100]: force_pangoro || is_pangolin(). Looks like not correct

File: node/cli/src/command.rs (hunk 1 above)
Comment (AurevoirXavier) [reply node/cli/src/command.rs:80]: Default to Pangoro (Darwinia).

File: node/service/src/chain_spec/template.rs [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+		genesis,
+		vec![],
+		None,
+		None,
+		None,
+		None,
+	)
+}
+
+fn get_authority_keys_from_seed(s: &str) -> (AuraId, GrandpaId) {
```
[the remark this answers is not in the document]
Comment (AurevoirXavier) [reply node/service/src/chain_spec/template.rs:106]: But only template use `AuraId`.

[the remark this answers is not in the document]
Comment (AsceticBear) [reply node/service/src/chain_spec/template.rs:106]: How about this way:

```rs
#[cfg(feature = "template")]
use sp_consensus_aura::sr25519::AuthorityId as AuraId;
#[cfg(feature = "template")]
fn get_authority_keys_from_seed(s: &str) -> (AuraId, GrandpaId) {
	(get_from_seed::<AuraId>(s), get_from_seed::<GrandpaId>(s))
}

```

[the remark this answers is not in the document]
Comment (AurevoirXavier) [reply node/service/src/chain_spec/template.rs:106]: But still, conflict (same name).
We need Pangoro and Pangolin, even enabling the template feature.

**Review by AsceticBear — APPROVED** [review]

**Pull request merged**